compiler/hie-core/src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -116,12 +151,19 @@ runLanguageServer options userHandlers getIdeState = do
 -- | Things that get sent to us, but we don't deal with.
 --   Set them to avoid a warning in VS Code output.
 setHandlersIgnore :: PartialHandlers
-setHandlersIgnore = PartialHandlers $ \_ x -> return x
-    {LSP.cancelNotificationHandler = none
-    ,LSP.initializedHandler = none
+setHandlersIgnore = PartialHandlers $ \WithMessage{..} x -> return x
+    {LSP.initializedHandler = none
     }
     where none = Just $ const $ return ()
 
+cancelHandler :: (LspId -> IO ()) -> PartialHandlers
+cancelHandler cancelRequest = PartialHandlers $ \_ x -> return x
+    {LSP.cancelNotificationHandler = Just $ \msg@NotificationMessage {_params} -> do
+            let reqId = case _params of CancelParams{_id} -> _id
```

Comment:
Why not just pattern match on the line above?

Reply:
Good point, changed.  I managed to confuse myself with `DuplicateRecordFields` here.